diff -NBaur actionlib/CMakeLists.txt actionlib_new/CMakeLists.txt
--- actionlib/CMakeLists.txt	2012-07-23 11:49:22.664771610 +0900
+++ actionlib_new/CMakeLists.txt	2012-07-23 11:55:13.316767637 +0900
@@ -6,20 +6,20 @@
   LIBRARIES actionlib
   INCLUDE_DIRS include
   )
-
+find_package(Boost REQUIRED COMPONENTS thread date_time)
 find_package(ROS REQUIRED COMPONENTS
   rostime roscpp_serialization roscpp_traits cpp_common # serialization stuff
   roscpp rosconsole xmlrpcpp #roscpp stuff
   actionlib_msgs rosunit rostest)
 
 include_directories(include)
-include_directories(${ROS_INCLUDE_DIRS})
+include_directories(${ROS_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
 
 # actionlib library
 add_library(actionlib src/connection_monitor.cpp
                       src/goal_id_generator.cpp)
 
-target_link_libraries(actionlib ${ROS_LIBRARIES})
+target_link_libraries(actionlib ${ROS_LIBRARIES} ${Boost_LIBRARIES})
 
 catkin_python_setup()
 
@@ -46,6 +46,9 @@
 add_executable(actionlib-add_two_ints_server EXCLUDE_FROM_ALL test/add_two_ints_server.cpp)
 target_link_libraries(actionlib-add_two_ints_server ${PROJECT_NAME} ${GTEST_LIBRARIES})
 
+add_executable(actionlib-add_two_ints_client EXCLUDE_FROM_ALL test/add_two_ints_client.cpp)
+target_link_libraries(actionlib-add_two_ints_client ${PROJECT_NAME} ${GTEST_LIBRARIES})
+
 add_executable(actionlib-ref_server EXCLUDE_FROM_ALL test/ref_server.cpp)
 target_link_libraries(actionlib-ref_server ${PROJECT_NAME})
 
