diff -NBaur -x .svn -x *.pyc ros/tools/rosclean/src/rosclean/__init__.py C:\work\src\ros/tools/rosclean/src/rosclean/__init__.py
--- ros/tools/rosclean/src/rosclean/__init__.py	Fri Apr 27 12:44:47 2012
+++ C:\work\src\ros/tools/rosclean/src/rosclean/__init__.py	Fri Apr 27 12:36:55 2012
@@ -75,7 +75,10 @@
 \trosclean check\tCheck usage of log files
 \trosclean purge\tRemove log files
 """)
-    sys.exit(os.EX_USAGE)
+    try:
+        sys.exit(os.EX_USAGE) # Unix only
+    except AttributeError:
+        sys.exit(1)
     
 def _get_check_dirs():
     home_dir = rospkg.get_ros_home()
diff -NBaur -x .svn -x *.pyc ros/tools/rosunit/scripts/check_test_ran.py C:\work\src\ros/tools/rosunit/scripts/check_test_ran.py
--- ros/tools/rosunit/scripts/check_test_ran.py	Fri Apr 27 12:44:46 2012
+++ C:\work\src\ros/tools/rosunit/scripts/check_test_ran.py	Fri Apr 27 12:36:17 2012
@@ -53,7 +53,10 @@
 \t%s --rostest pkg-name test-file.xml
 """%(NAME, NAME), file=sys.stderr)
     print(sys.argv)
-    sys.exit(os.EX_USAGE)
+    try:
+        sys.exit(os.EX_USAGE) # Unix only
+    except AttributeError:
+        sys.exit(1)
 
 def check_main():
     if len(sys.argv) < 2:
