diff -NBaur -x *.pyc -x .git rospkg/scripts/rosversion C:\work\src\rospkg/scripts/rosversion
--- rospkg/scripts/rosversion	Fri Apr 27 12:47:13 2012
+++ C:\work\src\rospkg/scripts/rosversion	Fri Apr 27 12:27:30 2012
@@ -84,4 +84,7 @@
         sys.stdout.write(version)
 else:
     sys.stderr.write(USAGE+'\n')
-    sys.exit(os.EX_USAGE)
+    try:
+        sys.exit(os.EX_USAGE)
+    except AttributeError:
+        sys.exit(1)
