diff -NBaur -x .svn ros_comm/tools/rosbag/include/rosbag/player.h ros_comm_total/tools/rosbag/include/rosbag/player.h
--- ros_comm/tools/rosbag/include/rosbag/player.h	2012-07-03 11:35:17.812154662 +0900
+++ ros_comm_total/tools/rosbag/include/rosbag/player.h	2012-07-03 11:20:35.136180479 +0900
@@ -36,9 +36,11 @@
 #define ROSBAG_PLAYER_H
 
 #include <sys/stat.h>
-#include <termios.h>
+#if !defined(_MSC_VER)
+  #include <termios.h>
+  #include <unistd.h>
+#endif
 #include <time.h>
-#include <unistd.h>
 
 #include <queue>
 #include <string>
@@ -176,8 +178,10 @@
 
     // Terminal
     bool    terminal_modified_;
+#if !defined(_MSC_VER)
     termios orig_flags_;
     fd_set  stdin_fdset_;
+#endif
     int     maxfd_;
 
     TimeTranslator time_translator_;
diff -NBaur -x .svn ros_comm/tools/rosbag/include/rosbag/recorder.h ros_comm_total/tools/rosbag/include/rosbag/recorder.h
--- ros_comm/tools/rosbag/include/rosbag/recorder.h	2012-07-03 11:35:17.812154662 +0900
+++ ros_comm_total/tools/rosbag/include/rosbag/recorder.h	2012-07-03 11:20:35.132180479 +0900
@@ -36,9 +36,11 @@
 #define ROSBAG_RECORDER_H
 
 #include <sys/stat.h>
-#include <termios.h>
+#if !defined(_MSC_VER)
+  #include <termios.h>
+  #include <unistd.h>
+#endif
 #include <time.h>
-#include <unistd.h>
 
 #include <queue>
 #include <string>
diff -NBaur -x .svn ros_comm/tools/rosbag/src/bag.cpp ros_comm_total/tools/rosbag/src/bag.cpp
--- ros_comm/tools/rosbag/src/bag.cpp	2012-07-03 11:35:17.808154659 +0900
+++ ros_comm_total/tools/rosbag/src/bag.cpp	2012-07-03 11:20:35.132180479 +0900
@@ -30,7 +30,11 @@
 #include "rosbag/query.h"
 #include "rosbag/view.h"
 
-#include <inttypes.h>
+#if defined(_MSC_VER)
+  #include <stdint.h> // only on v2010 and later -> is this enough for msvc and linux?
+#else
+  #include <inttypes.h>
+#endif
 #include <signal.h>
 
 #include <iomanip>
diff -NBaur -x .svn ros_comm/tools/rosbag/src/chunked_file.cpp ros_comm_total/tools/rosbag/src/chunked_file.cpp
--- ros_comm/tools/rosbag/src/chunked_file.cpp	2012-07-03 11:35:17.808154659 +0900
+++ ros_comm_total/tools/rosbag/src/chunked_file.cpp	2012-07-03 11:20:35.128180479 +0900
@@ -39,6 +39,18 @@
 #include <boost/format.hpp>
 
 #include <ros/ros.h>
+#ifdef _WIN32
+#    ifdef __MINGW32__
+#      define fseeko fseeko64
+#      define ftello ftello64
+//     not sure if we need a ftruncate here yet or not
+#    else
+#        define fseeko _fseeki64
+#        define ftello _ftelli64
+#        include <io.h>
+#        define ftruncate _chsize
+#    endif
+#endif
 
 using std::string;
 using boost::format;
diff -NBaur -x .svn ros_comm/tools/rosbag/src/player.cpp ros_comm_total/tools/rosbag/src/player.cpp
--- ros_comm/tools/rosbag/src/player.cpp	2012-07-03 11:35:17.808154659 +0900
+++ ros_comm_total/tools/rosbag/src/player.cpp	2012-07-03 11:20:35.132180479 +0900
@@ -36,7 +36,9 @@
 #include "rosbag/message_instance.h"
 #include "rosbag/view.h"
 
-#include <sys/select.h>
+#if !defined(_MSC_VER)
+  #include <sys/select.h>
+#endif
 
 #include <boost/foreach.hpp>
 #include <boost/format.hpp>
@@ -401,6 +403,7 @@
 	if (terminal_modified_)
 		return;
 
+#if !defined(_MSC_VER)
     const int fd = fileno(stdin);
     termios flags;
     tcgetattr(fd, &orig_flags_);
@@ -413,7 +416,7 @@
     FD_ZERO(&stdin_fdset_);
     FD_SET(fd, &stdin_fdset_);
     maxfd_ = fd + 1;
-
+#endif
     terminal_modified_ = true;
 }
 
@@ -421,9 +424,10 @@
 	if (!terminal_modified_)
 		return;
 
+#if !defined(_MSC_VER)
     const int fd = fileno(stdin);
     tcsetattr(fd, TCSANOW, &orig_flags_);
-
+#endif
     terminal_modified_ = false;
 }
 
@@ -431,16 +435,17 @@
 #ifdef __APPLE__
     fd_set testfd;
     FD_COPY(&stdin_fdset_, &testfd);
-#else
+#elif !defined(_MSC_VER)
     fd_set testfd = stdin_fdset_;
 #endif
 
+#if !defined(_MSC_VER)
     timeval tv;
     tv.tv_sec  = 0;
     tv.tv_usec = 0;
     if (select(maxfd_, &testfd, NULL, NULL, &tv) <= 0)
         return EOF;
-
+#endif
     return getc(stdin);
 }
 
diff -NBaur -x .svn ros_comm/tools/rosbag/src/record.cpp ros_comm_total/tools/rosbag/src/record.cpp
--- ros_comm/tools/rosbag/src/record.cpp	2012-07-03 11:35:17.808154659 +0900
+++ ros_comm_total/tools/rosbag/src/record.cpp	2012-07-03 11:20:35.128180479 +0900
@@ -147,9 +147,9 @@
       if ((iss >> duration).fail())
         throw ros::Exception("Duration must start with a floating point number.");
 
-      if (!iss.eof() and ((iss >> unit).fail()))
+      if ( (!iss.eof() && ((iss >> unit).fail())) ) {
         throw ros::Exception("Duration unit must be s, m, or h");
-      
+      }
       if (unit == std::string(""))
         multiplier = 1.0;
       else if (unit == std::string("s"))
diff -NBaur -x .svn ros_comm/tools/rosbag/src/recorder.cpp ros_comm_total/tools/rosbag/src/recorder.cpp
--- ros_comm/tools/rosbag/src/recorder.cpp	2012-07-03 11:35:17.808154659 +0900
+++ ros_comm_total/tools/rosbag/src/recorder.cpp	2012-07-03 11:20:35.128180479 +0900
@@ -35,7 +35,9 @@
 #include "rosbag/recorder.h"
 
 #include <sys/stat.h>
-#include <sys/statvfs.h>
+#if !defined(_MSC_VER)
+  #include <sys/statvfs.h>
+#endif
 #include <time.h>
 
 #include <queue>
@@ -574,6 +576,7 @@
 }
 
 bool Recorder::checkDisk() {
+#if !defined(_MSC_VER)
     struct statvfs fiData;
     if ((statvfs(bag_.getFileName().c_str(), &fiData)) < 0) {
         ROS_WARN("Failed to check filesystem stats.");
@@ -593,6 +596,9 @@
     else {
         writing_enabled_ = true;
     }
+#else
+    writing_enabled_ = true;
+#endif
 
     return true;
 }
