diff -NBaur -x .svn -x .hg rospack/CMakeLists.txt C:\work\src\rospack/CMakeLists.txt
--- rospack/CMakeLists.txt	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/CMakeLists.txt	Mon May  7 12:12:55 2012
@@ -14,10 +14,10 @@
   message(STATUS "Rospack building shared objects.")
 else()
   message(STATUS "Rospack building static objects.")
-  add_definitions(-DROS_BUILD_STATIC_LIBS) # tinyxml.h:86
+  add_definitions(-DROS_BUILD_STATIC_LIBS) # tinyxml.h:86,  macros.h
 endif()
 
-add_definitions(-Wall)
+#add_definitions(-Wall)
 
 set(API_BACKCOMPAT_V1 "YES"
     CACHE BOOL "Whether to enable backwards compatibility with old C++ API")
diff -NBaur -x .svn -x .hg rospack/include/rospack/macros.h C:\work\src\rospack/include/rospack/macros.h
--- rospack/include/rospack/macros.h	Thu Jan  1 09:00:00 1970
+++ C:\work\src\rospack/include/rospack/macros.h	Mon May  7 12:40:00 2012
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2008, Willow Garage, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the names of Stanford University or Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef ROSPACK_MACROS_H_
+#define ROSPACK_MACROS_H_
+
+/*
+ * Windows macros for dll exports
+ *
+ * This replicates what ros has in roscpp_core/cpp_common/macros.h
+ * so rospack can be standalone.
+ *
+ * See also http://gcc.gnu.org/wiki/Visibility
+ */
+
+#if defined(_MSC_VER)
+    #define ROS_HELPER_IMPORT __declspec(dllimport)
+    #define ROS_HELPER_EXPORT __declspec(dllexport)
+    #define ROS_HELPER_LOCAL
+#elif __GNUC__ >= 4
+    #define ROS_HELPER_IMPORT __attribute__ ((visibility("default")))
+    #define ROS_HELPER_EXPORT __attribute__ ((visibility("default")))
+    #define ROS_HELPER_LOCAL  __attribute__ ((visibility("hidden")))
+#else
+    #define ROS_HELPER_IMPORT
+    #define ROS_HELPER_EXPORT
+    #define ROS_HELPER_LOCAL
+#endif
+
+// Ignore warnings about import/exports when deriving from std classes.
+#ifdef _MSC_VER
+  #pragma warning(disable:4251)
+  #pragma warning(disable:4275)
+  #pragma warning(disable:4514) // unreferenced inline function has been removed
+  #pragma warning(disable:4668) // warnigns about used, but not defined macros, replacing with 0
+  #pragma warning(disable:4710) // function not inlined
+  #pragma warning(disable:4738) // storing floats, memory splitting, possible loss in performance
+  #pragma warning(disable:4820) // structs and padding
+#endif
+
+#ifdef ROS_BUILD_STATIC_LIBS // ros is being built around shared libraries
+  #define ROSPACK_DECL
+#else // ros is being built around shared libraries
+  #ifdef rospack_EXPORTS // we are building a shared lib/dll
+    #define ROSPACK_DECL ROS_HELPER_EXPORT
+  #else // we are using shared lib/dll
+    #define ROSPACK_DECL ROS_HELPER_IMPORT
+  #endif
+#endif
+
+#endif /* ROSPACK_MACROS_H_ */
diff -NBaur -x .svn -x .hg rospack/include/rospack/rospack.h C:\work\src\rospack/include/rospack/rospack.h
--- rospack/include/rospack/rospack.h	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/include/rospack/rospack.h	Mon May  7 12:14:49 2012
@@ -111,6 +111,7 @@
 #include <vector>
 #include <set>
 #include <list>
+#include "macros.h"
 
 //#ifdef ROSPACK_API_BACKCOMPAT_V1
 #if 1 // def ROSPACK_API_BACKCOMPAT_V1
@@ -136,7 +137,7 @@
  * use the functionality provided here through one of the derived classes,
  * Rosstack or Rospack.
  */
-class Rosstackage
+class ROSPACK_DECL Rosstackage
 {
   private:
     std::string manifest_name_;
@@ -485,7 +486,7 @@
  * @brief Package crawler.  Create one of these to operate on a package
  * tree.  Call public methods inherited from Rosstackage.
  */
-class Rospack : public Rosstackage
+class ROSPACK_DECL Rospack : public Rosstackage
 {
   public:
     /**
@@ -503,7 +504,7 @@
  * @brief Stack crawler.  Create one of these to operate on a stack
  * tree.  Call public methods inherited from Rosstackage.
  */
-class Rosstack : public Rosstackage
+class ROSPACK_DECL Rosstack : public Rosstackage
 {
   public:
     /**
diff -NBaur -x .svn -x .hg rospack/include/rospack/rospack_backcompat.h C:\work\src\rospack/include/rospack/rospack_backcompat.h
--- rospack/include/rospack/rospack_backcompat.h	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/include/rospack/rospack_backcompat.h	Mon May  7 12:15:13 2012
@@ -30,6 +30,7 @@
 #define ROSPACK_ROSPACK_BACKCOMPAT_H
 
 #include <string>
+#include "macros.h"
 
 namespace rospack
 {
@@ -38,7 +39,7 @@
  * @brief Backward compatibility API for librospack (DEPRECATED).
  * @deprecated Used by roslib.  Don't use in new code.
  */
-class ROSPack
+class ROSPACK_DECL ROSPack
 {
   private:
     std::string output_;
diff -NBaur -x .svn -x .hg rospack/src/rospack.cpp C:\work\src\rospack/src/rospack.cpp
--- rospack/src/rospack.cpp	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/src/rospack.cpp	Mon May  7 12:14:56 2012
@@ -36,14 +36,20 @@
 #if defined(WIN32)
   #include <windows.h>
   #include <direct.h>
-  #include <libgen.h> // for dirname
   #include <fcntl.h>  // for O_RDWR, O_EXCL, O_CREAT
-#else //!defined(WIN32)
-  #include <sys/types.h>
-  #include <libgen.h>
-  #include <pwd.h>
-  #include <unistd.h>
-  #include <sys/time.h>
+  // simple workaround - could have issues though. See
+  //   http://stackoverflow.com/questions/2915672/snprintf-and-visual-studio-2010
+  // for potentially better solutions. Similar probably applies for some of the others
+  #define snprintf _snprintf
+  #define pclose _pclose
+  #define popen _popen
+  #define PATH_MAX MAX_PATH
+  #if defined(__MINGW32__)
+    #include <libgen.h> // for dirname
+  #endif
+  #if defined(_MSC_VER)
+    #include <io.h> // _mktemp_s
+  #endif
 #endif
 
 #include <climits>
diff -NBaur -x .svn -x .hg rospack/src/rospack_cmdline.cpp C:\work\src\rospack/src/rospack_cmdline.cpp
--- rospack/src/rospack_cmdline.cpp	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/src/rospack_cmdline.cpp	Mon May  7 12:38:53 2012
@@ -27,6 +27,7 @@
 
 #include "rospack/rospack.h"
 #include "utils.h"
+#include "rospack_cmdline.h"
 
 #include <boost/program_options.hpp>
 #include <boost/algorithm/string.hpp>
@@ -37,15 +38,13 @@
 
 namespace po = boost::program_options;
 
-namespace rospack
-{
+namespace rospack {
 
-bool parse_args(int argc, char** argv, 
+bool parse_args(int argc, char** argv,
                 rospack::Rosstackage& rp,
                 po::variables_map& vm);
 
-bool
-rospack_run(int argc, char** argv, rospack::Rosstackage& rp, std::string& output)
+bool rospack_run(int argc, char** argv, rospack::Rosstackage& rp, std::string& output)
 {
   po::variables_map vm;
 
@@ -792,7 +791,7 @@
 }
 
 bool
-parse_args(int argc, char** argv, 
+parse_args(int argc, char** argv,
            rospack::Rosstackage& rp, po::variables_map& vm)
 {
   po::options_description desc("Allowed options");
@@ -825,4 +824,4 @@
   return true;
 }
 
-}
+} // namespace rospack
diff -NBaur -x .svn -x .hg rospack/src/rospack_cmdline.h C:\work\src\rospack/src/rospack_cmdline.h
--- rospack/src/rospack_cmdline.h	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/src/rospack_cmdline.h	Mon May  7 12:27:51 2012
@@ -28,12 +28,13 @@
 #ifndef ROSPACK_ROSPACK_CMDLINE_H
 #define ROSPACK_ROSPACK_CMDLINE_H
 
+#include "../include/rospack/macros.h"
 #include "rospack/rospack.h"
 
 namespace rospack
 {
 
-bool rospack_run(int argc, char** argv, 
+ROSPACK_DECL bool rospack_run(int argc, char** argv,
                  rospack::Rosstackage& rp, 
                  std::string& output);
 
diff -NBaur -x .svn -x .hg rospack/src/utils.cpp C:\work\src\rospack/src/utils.cpp
--- rospack/src/utils.cpp	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/src/utils.cpp	Mon May  7 12:40:57 2012
@@ -29,7 +29,8 @@
 #include <vector>
 #include <boost/algorithm/string.hpp>
 #include <boost/tr1/unordered_set.hpp>
-        
+#include "utils.h"
+
 namespace rospack
 {
 
diff -NBaur -x .svn -x .hg rospack/src/utils.h C:\work\src\rospack/src/utils.h
--- rospack/src/utils.h	Thu May  3 18:27:02 2012
+++ C:\work\src\rospack/src/utils.h	Mon May  7 12:40:28 2012
@@ -28,6 +28,7 @@
 #ifndef ROSPACK_UTILS_H
 #define ROSPACK_UTILS_H
 
+#include "../include/rospack/macros.h"
 #include <string>
 
 namespace rospack
@@ -37,11 +38,11 @@
 static const char* ROSPACK_NAME = "rospack";
 static const char* ROSSTACK_NAME = "rosstack";
 
-void deduplicate_tokens(const std::string& instring, 
+ROSPACK_DECL void deduplicate_tokens(const std::string& instring,
                         bool last,
                         std::string& outstring);
 
-void
+ROSPACK_DECL void
 parse_compiler_flags(const std::string& instring, 
                      const std::string& token,
                      bool select,
