diff -NBaur roscpp_core/rostime/CMakeLists.txt roscpp_core_chrono/rostime/CMakeLists.txt
--- roscpp_core/rostime/CMakeLists.txt	2012-08-08 22:29:26.260933564 +1000
+++ roscpp_core_chrono/rostime/CMakeLists.txt	2012-08-10 22:25:13.328524661 +1000
@@ -4,7 +4,13 @@
 
 find_package(cpp_common REQUIRED)
 include_directories(${cpp_common_INCLUDE_DIRS})
-find_package(Boost REQUIRED COMPONENTS date_time thread)
+
+# Boost 1.50 thread needs chrono                                                      
+find_package(Boost 1.50.0 COMPONENTS date_time thread chrono)
+if(NOT Boost_FOUND)
+  find_package(Boost REQUIRED COMPONENTS date_time thread)
+endif()
+
 
 add_library(rostime 
   src/time.cpp src/rate.cpp src/duration.cpp)
