diff -NBaur roscpp_core/rostime/src/time.cpp roscpp_core_rostime/rostime/src/time.cpp
--- roscpp_core/rostime/src/time.cpp	2012-04-26 12:16:42.102597524 +0900
+++ roscpp_core_rostime/rostime/src/time.cpp	2012-04-26 12:17:40.326596865 +0900
@@ -122,8 +122,8 @@
     // problems in Windows Vista, and this API is by far the most accurate that
     // I know of in Windows, so I'll use it here despite all these caveats
     static LARGE_INTEGER cpu_freq, init_cpu_time;
-    uint32_t start_sec = 0;
-    uint32_t start_nsec = 0;
+    static uint32_t start_sec = 0;
+    static uint32_t start_nsec = 0;
     if ( ( start_sec == 0 ) && ( start_nsec == 0 ) )
       {
         QueryPerformanceFrequency(&cpu_freq);
