diff -uNr --exclude=.svn --exclude=build --exclude=lib rostime.original/src/time.cpp rostime.working_copy/src/time.cpp
--- rostime.original/src/time.cpp	2011-12-27 16:57:14.621223827 +0900
+++ rostime.working_copy/src/time.cpp	2011-12-27 17:03:22.537736039 +0900
@@ -121,8 +121,8 @@
     // problems in Windows Vista, and this API is by far the most accurate that
     // I know of in Windows, so I'll use it here despite all these caveats
     static LARGE_INTEGER cpu_freq, init_cpu_time;
-    uint32_t start_sec = 0;
-    uint32_t start_nsec = 0;
+    static uint32_t start_sec = 0;
+    static uint32_t start_nsec = 0;
     if ( ( start_sec == 0 ) && ( start_nsec == 0 ) )
       {
         QueryPerformanceFrequency(&cpu_freq);
