--- svn.original.py	Mon Mar  5 09:59:26 2012
+++ svn_ssl.py	Mon Mar  5 14:01:06 2012
@@ -58,6 +58,9 @@
 
 class SvnClient(VcsClientBase):
 
+    def ssl_server_trust_prompt(self, trust_dict):
+        return True, trust_dict['failures'], True
+
     def __init__(self, path):
         """
         :raises: VcsError if python-svn not detected
@@ -75,6 +78,7 @@
         except:
             raise VcsError("svn not installed")
         self._pysvnclient = pysvn.Client()
+        self._pysvnclient.callback_ssl_server_trust_prompt = self.ssl_server_trust_prompt
 
     @staticmethod
     def get_environment_metadata():
