--- genutil.original.py	2011-12-26 16:59:16.069262827 +0900
+++ genutil.py	2011-12-26 16:56:07.252970973 +0900
@@ -166,7 +166,7 @@
         elif not os.path.isdir(basedir):
             raise self.exception("file preventing the creating of module directory: %s"%dir)
         p = os.path.join(basedir, '__init__.py')
-        f = open(p, 'w')
+        f = open(p, 'a')
         try:
             #this causes more problems than anticipated -- for pure python
             #packages it works fine, but in C++ packages doxygen seems to prefer python first.
